crates/dbs-allocator/src/interval_tree.rs

```diff
@@ -1033,37 +1033,91 @@ mod tests {
 
     #[should_panic]
     #[test]
-    fn test_tree_insert_intersect() {
+    fn test_tree_insert_intersect_on_right() {
         let mut tree = IntervalTree::<u64>::new();
         tree.insert(Range::new(0x100, 0x200u32), Some(1));
         tree.insert(Range::new(0x200, 0x2ffu64), None);
     }
 
+    #[should_panic]
+    #[test]
+    fn test_tree_insert_intersect_on_left() {
+        let mut tree = IntervalTree::<u64>::new();
+        tree.insert(Range::new(0x100, 0x200u32), Some(1));
+        tree.insert(Range::new(0x000, 0x100u64), None);
+    }
+
     #[test]
     fn test_tree_get_superset() {
         let mut tree = IntervalTree::<u64>::new();
-        tree.insert(Range::new(0x100u64, 0x100u64), Some(1));
-        tree.insert(Range::new(0x200, 0x2ffu64), None);
+        tree.insert(Range::new(0x100u32, 0x100u32), Some(1));
+        tree.insert(Range::new(0x001u16, 0x008u16), None);
```

Comment:
Why do you change the type of number from u64 to u32/u16?

Reply:
Various unsigned integer types are used in previously existing UTs, so I continue to use it to cover the functions written under templates.